qemu/tests/block_commit_stress.py

```diff
@@ -0,0 +1,132 @@
+            session.sendline(cmd)
+            if not self.app_running():
+                return True
+            return False
+
+        error.context("stop stress app in guest", logging.info)
+        stopped = utils_misc.wait_for(_unload_stress, first=2.0,
+                                      text="wait stress app quit",
+                                      step=1.0, timeout=params["wait_timeout"])
+        if not stopped:
+            raise logging.warn("stress app is still running")
+
+    def app_running(self):
+        """
+        check stress app really run in background
+        """
+        session = self.get_session()
+        cmd = self.params.get("check_cmd")
+        status = session.cmd_status(cmd, timeout=120)
+        return status == 0
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    block_commit_stress test:
+    1). load stress in guest
+    2). create live snapshot base->sn1->sn2->sn3->sn4
+    3). merge sn3 to sn1(sn3->sn1), the snapshot chain should be base->sn1->sn4
+    4). commit block device and wait to finished
+    5). check backing file after commit
+    6). quit stress app
+    7). reboot and verify guest can response correctly
+
+    :param test: Kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    timeout = int(params.get("login_timeout", 360))
+    session = vm.wait_for_login(timeout=timeout)
+    alive_check_cmd = params.get("alive_check_cmd", "dir")
+    image_file = storage.get_image_filename(params, data_dir.get_data_dir())
+    image_dir = os.path.dirname(image_file)
+    qemu_img = utils_misc.get_qemu_img_binary(params)
+    speed = int(params.get("limited_speed", 0))
+    wait_timeout = int(params.get("wait_timeout", 3600))
+
+    def wait_job_done(timeout=3600):
+        """
+        Wait for job on the device done, raise TestFail exception if timeout
+        """
+        if utils_misc.wait_for(lambda:
+                               not vm.monitor.query_block_job(device_id),
+                               timeout, first=0.2, step=2.0,
+                               text="Wait for canceling block job") is None:
+            raise error.TestFail("Wait job finish timeout in %ss" % timeout)
+
+    def verify_backingfile(except_backingfile):
+        """
+        Got backingfile from monitor then verfiy it except_backingfile,
+        if not raise TestFail exception
+        """
+        backing_file = vm.monitor.get_backingfile(device_id)
+        if backing_file != except_backingfile:
+            msg = "expected backingfile: %s " % except_backingfile
+            msg += "actual backingfile: %s " % backing_file
+            logging.info(msg)
+            raise error.TestFail("Unexpect backingfile(%s)" % backing_file)
+
+    tag = params.get("source_image", "image1")
+    stress_test = BlockCommitStress(test, params, env, tag)
```

Comment:
seems you don't use stress_test, and I didn't find anywhere call action_when_start(), so your definition for "before_start" in cfg file wont be execute, which means, load_stress wont be execute, so does after_finish steps